kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/AchPayeeBankAcctInputFileType.java

```diff
@@ -0,0 +1,934 @@
+			reportLines.add(reportLine);
+			validAcct = false;
+		}
+		else {
+			payeeAcct.setPayeeName(payee.getNameUnmasked().toUpperCase());
+		}
+		
+		if (StringUtils.isBlank(achTransactionType)) {
+			payeeAcct.setAchTransactionType(KFSConstants.ACHFileDefaultConstants.DEFAULT_ACH_TRANSACTION_TYPE);
+		}
+		
+		if (StringUtils.isBlank(bankAcctTypeCode)) {
+			reportLine = new PayeeReportLine(payeeIdNumber, payeeName, KFSConstants.ACHFileConstants.ACH_PAYEE_ACCT_MISSING_BANK_ACCT_TYP_ERROR);
+			reportLines.add(reportLine);
+			validAcct = false;
+		}
+		else if (!validAcctTypeCodes.contains(bankAcctTypeCode)) {
+			reportLine = new PayeeReportLine(payeeIdNumber, payeeName, KFSConstants.ACHFileConstants.ACH_PAYEE_ACCT_INVALID_BANK_ACCT_TYP_ERROR);
+			reportLines.add(reportLine);
+			validAcct = false;
+		}
+		
+		return validAcct;
+	}
+	
+	protected void writeReport(List<PayeeReportLine> lines, String systemSource) {
+		File reportFile = getFile(reportPath, reportPrefix, reportExtension, true);
+		BufferedWriter writer = null;
+		String payeeId;
+		String payeeNm;
+		
+		try {
+			LOG.debug("reportFile: " + reportFile.getAbsolutePath());
+			
+			writer = new BufferedWriter(new PrintWriter(reportFile));
+			writeReportTitle(writer, systemSource);
+			writeReportHeadings(writer);
+			for (PayeeReportLine line : lines) {
+				payeeId = rightPad(line.getPayeeID(), 20, " ");
+				payeeNm = rightPad(line.getPayeeNm(), 35, " ");
+				writer.write(payeeId + payeeNm + line.getMessage());
+				writer.newLine();
+				writer.newLine();
+			}
+		}
+		catch (FileNotFoundException e) {
+			LOG.error(reportFile + " not found " + " " + e.getMessage());
+			throw new RuntimeException(reportFile + " not found " + " " + e.getMessage(), e);
+		}
+		catch (IOException e) {
+			LOG.error("Error writing to BufferedWriter " + e.getMessage());
+			throw new RuntimeException("Error writing to BufferedWriter " + e.getMessage(), e);
+		}
+		finally {
+			try {
+				writer.flush();
+				writer.close();
+			}
+			catch (Exception e) {
+				LOG.error("Error encountered when closing BufferedWriter");
+			}
+		}
+	}
+	
+	protected void writeReportTitle(BufferedWriter writer, String systemSource) {
+		OriginationCode originCd = originationCodeDao.findByCode(systemSource);
+		String source = systemSource + " - Unknown System Source";
+		if (originCd != null) {
+			source = originCd.getFinancialSystemDatabaseDesc();
+		}
+		
+		try {
+			writer.write(dateTimeService.toDateString(dateTimeService.getCurrentDate()));
+			writer.newLine();
+			writer.newLine();
+			writer.write("Payee ACH Account Interface");
+			writer.newLine();
+			writer.write("System Source: " + systemSource + " - " + source);
+			writer.newLine();
+			writer.newLine();
+		}
+		catch (IOException e) {
+			LOG.error("Error writing to BufferedWrtier " + e.getMessage());
+			throw new RuntimeException("Error writing to BufferedWriter " + e.getMessage(), e);
+		}
```

Comment:
This writer is being passed to writeReportTitle() by writeReport(). The writer will be closed at writeReport().